### DISCUSSION TITLE
Using without SwiftUI

### DISCUSSION THREAD

**Discussion by martindufort** [danielsaidi/RichTextKit#164] [General]
I'm wondering if we are losing any of the RichTextKit functionalities if we are integrating it with an AppKit/UIKit only codebase. 

Any parts of the framework that is SwiftUI-only?

**Comment by danielsaidi** [comment]
Hi @martindufort 

Much of the underlying functionality is cross-platform, but a lot of utilities are SwiftUI only.

**Comment by martindufort** [comment]
OK so here, in our company land, we are still AppKit / UIKit. 
So what's our best strategy to leverage the complete functional set of RichTextKit. 

Should we go `UIHostingController` as a child view controller and set `RichTextEditor` as the `rootView`?
Anyone here tested this path?

Thanks

**Comment by danielsaidi** [comment]
Hi @martindufort 

I'm not really sure what the best approach would be. I tried to keep the SwiftUI layer thin, and just make it call down to the platform-specific layers, but it quickly became too brittle.

You can perhaps check if the rich text component APIs are enough to get you where you need.
